system/core/tags.php

```diff
@@ -0,0 +1,192 @@
+<?php
+	// TODO:
+	// - add styling for default theme
+	// - implement error notification if new tag exists
+	// - implement tag management page: edit, translate, remove or merge tags
+	// - implement tagcontainer pagetype
+
+	function tagList(HyphaDomElement $pageListNode, $lang) {
+		/** @var HyphaDomElement $hyphaXml */
+		global $hyphaXml;
+
+		$selectedTagIds = [];
+
+		/** @var HyphaDomElement $tagEl */
+		// Build list of active tags for given page
+		foreach ($pageListNode->findXPath('tag') as $tagEl) $selectedTagIds[] = $tagEl->getAttribute('id');
+
+		$selectedTags = [];
+		$deselectedTags = [];
+
+		$lang = htmlspecialchars($lang);
```

Comment:
This seems like the wrong place to encode `$lang`. Ususally, you want to encode things as late as possible, just before inserting into (in this case) HTML. Here, `$lang` is encoded to early, and you can see it is then used in an xpath expression (where it should *not* be HTML-encoded, just Xpath-encoded). In fact, it seems that `$lang` is *only* used as part of xpath, never inserted in HTML, so I think this line can simply be dropped?